lib/account-manager.js

```diff
@@ -18,34 +16,124 @@
  */
 
 /**
+ * The real record that's stored.
+ *
  * @typedef {Object} accountManager~record
  * @property {string} email
  * @property {string} passwordHash
  * @property {accountManager~passwordHashConfig} passwordHashConfig
- * @property {Object} totp
- * @property {string} totp.key
+ * @property {Object} mfa
+ * @property {Object} mfa.totp
+ * @property {string} mfa.totp.key
+ */
+
+/**
+ * The public version of what's stored, along with other data that may be
+ * useful for routes
+ *
+ * @typedef {Object} accountManager~recordResponse
+ * @property {string} id
+ * @property {string} login New login cookie
+ * @property {Object} record
+ * @property {challengeManager~challengeHashConfig} record.challengeHashConfig
+ * @property {string} record.email
+ * @property {accountManager~passwordHashConfig} record.passwordHashConfig
+ */
+
+/**
+ * Information required for attempting a login.
+ *
+ * @typedef {Object} accountManager~loginHashConfigRequest
+ * @property {configManager~challengeHashConfig} challengeHashConfig
+ * @property {accountManager~passwordHashConfig} passwordHashConfig
+ */
+
+/**
+ * A request for logging in.  Validates the challenge and the MFA.
+ *
+ * @typedef {Object} accountManager~loginRequest
+ * @property {string} challengeHash
+ * @see {@link ../../schema/account/login-request.json}
  */
 
-module.exports = (config, promise, random, storageServiceFactory) => {
+/**
+ * Successful logins return the session ID.  This must be provided as a cookie
+ * in future requests.
+ *
+ * @typedef {Object} accountManager~loginResponse
+ * @property {string} sessionId
+ */
+
+module.exports = (base64, challengeManager, config, OtDate, promise, random, record, sessionManager, storageServiceFactory, totp) => {
     var accountIdLength, passwordHashConfigBase, passwordSaltLength, storageService;
 
 
     /**
      * Create a new account record
      *
-     * @param {accountManager~record} record
+     * @param {accountManager~record} recordData
      * @return {Promise.<string>} account ID
      */
-    function createAsync(record) {
+    function createAsync(recordData) {
         return random.idAsync(accountIdLength).then((id) => {
-            return storageService.putAsync(id, record, {
-                email: record.email
+            return storageService.putAsync(id, recordData, {
+                email: recordData.email
             }).then(() => {
                 return id;
             });
         });
     }
 
+
+    /**
+     * Verify the password and MFA.  If login is successful, return
+     * information for the cookie.
+     *
+     * @param {string} id Account ID
```

Comment:
Why not name this argument `accountId` since it is named that way in other methods?